Tools/uorb_graph/create.py

```diff
@@ -311,53 +404,89 @@ def _build_recursive(self, path):
         cmake_file = 'CMakeLists.txt'
         new_module = False
         if cmake_file in entries:
-            new_module = self._extract_module_name(os.path.join(path, cmake_file))
+            new_module = self._extract_build_information(os.path.join(path, cmake_file), **kwargs)
 
         # iterate directories recursively
         for entry in entries:
             file_name = os.path.join(path, entry)
             if os.path.isdir(file_name):
-                self._build_recursive(file_name)
+                self._build_recursive(file_name, **kwargs)
 
 
         # iterate source files
-        # Note: we could skip the entries if we're not in a module, but we don't
-        # so that we get appropriate error messages to know where we miss subs
-        # or pubs
-        for entry in entries:
-            file_name = os.path.join(path, entry)
-            if os.path.isfile(file_name):
-                _, ext = os.path.splitext(file_name)
-                if ext in ['.cpp', '.c', '.h', '.hpp']:
-                    self._process_source_file(file_name)
+        # Note: Skip all entries if we're not in a module -- both finding known pubs/subs and emitting warnings
```

Comment:
Is this correct?
What about libraries if we have a scope whitelist?

Reply:
Okay, (a) outdated comment -- it should actually be `if self._in_scope()`.

That compound if statement was actually in error -- we shouldn't be filtering on the whitelist at this point -- just if we're in _any_ scope at all. 

Will update this.

Reply:
Why not keep the existing behavior? Fail if not in a scope.

Reply:
Per request, the behavior is changed to warn about line if an ORB_ID was found outside of a containing scope (module or library)

But academic anyway, because I couldn't find any examples of this.

Reply:
It's not academic as it will help to raise issues in future refactorings, instead of silently breaking things. It adds robustness and therefore I wanted it to be an assertion.
Especially since this will run as CI and warnings can go unnoticed for quite some time.

Reply:
Okay I see your point. I've made this change on #17175 .